File: frontend/src/types/lostItem.ts
export type LostItemStatus = "STILL_LOOKING" | "FOUND" | "CLAIMED";

export interface LostItem {
  itemId: number;
  itemName: string;
  itemDesc: string;
  itemLocation: string;
  founderNumber: string;
  status: LostItemStatus;
  createdByUserId: number;
}

export interface LostItemForm {
  itemName: string;
  itemDesc: string;
  itemLocation: string;
  founderNumber: string;
  status: LostItemStatus;
}

export interface LostItemStatusHistoryEntry {
  statusChangeId: number;
  fromStatus: LostItemStatus | null;
  toStatus: LostItemStatus;
  changedByUserId: number;
  changedAt: string;
}

export interface LostItemDetails {
  item: LostItem;
  statusHistory: LostItemStatusHistoryEntry[];
}

File: backend/src/main/java/tech/bhos/Lost_Items/model/LostItem.java
package tech.bhos.Lost_Items.model;

import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.PrePersist;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.stereotype.Component;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Component
@Entity
public class LostItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer itemId;

    private String itemName;
    private String itemDesc;
    private String itemLocation;
    private String founderNumber;
    @Enumerated(EnumType.STRING)
    private LostItemStatus status;
    private Long createdByUserId;

    @PrePersist
    public void ensureDefaults() {
        if (status == null) {
            status = LostItemStatus.STILL_LOOKING;
        }
    }
}
